config/types/node.go

```diff
@@ -26,35 +26,43 @@ var (
 	ErrFileIllegalMode = errors.New("illegal file mode")
 )
 
-// Node represents all common info for files (special types, e.g. directories, included).
-type Node struct {
-	Filesystem string    `json:"filesystem,omitempty"`
-	Path       Path      `json:"path,omitempty"`
-	Mode       NodeMode  `json:"mode,omitempty"`
-	User       NodeUser  `json:"user,omitempty"`
-	Group      NodeGroup `json:"group,omitempty"`
-}
-
-type NodeUser struct {
-	Id int `json:"id,omitempty"`
+func (n Node) ValidateFilesystem() report.Report {
+	r := report.Report{}
+	if n.Filesystem == "" {
+		r.Add(report.Entry{
+			Message: ErrNoFilesystem.Error(),
+			Kind:    report.EntryError,
+		})
+	}
+	return r
 }
 
-type NodeGroup struct {
-	Id int `json:"id,omitempty"`
+func (n Node) ValidateMode() report.Report {
+	r := report.Report{}
+	if (n.Mode &^ 07777) != 0 {
```

Comment:
Can we just use '>' here?

I'm pretty sure it's easier to think "the mode is too big" than "the mode has high bits that don't get and-xor'd out"